ui/packages/catalog-ui-search/src/main/webapp/component/search-form/search-form.view.js

```diff
@@ -46,6 +46,7 @@
         }
     },
     changeView: function() {
+        let oldType;
```

Comment:
Should we just add `let oldType = this.options.queryModel.get('type');` . here and remove all the case-statement reassignments?